hummingbot/market/bitfinex/bitfinex_market.pyx

```diff
@@ -0,0 +1,1161 @@
+        """
+        if self._order_tracker_task is not None:
+            self._order_tracker_task.cancel()
+        if self._status_polling_task is not None:
+            self._status_polling_task.cancel()
+        if self._user_stream_tracker_task is not None:
+            self._user_stream_tracker_task.cancel()
+        if self._user_stream_event_listener_task is not None:
+            self._user_stream_event_listener_task.cancel()
+        self._order_tracker_task = self._status_polling_task = self._user_stream_tracker_task = \
+            self._user_stream_event_listener_task = None
+
+    async def _status_polling_loop(self):
+        """
+        Background process that periodically pulls for changes from the rest API
+        """
+        while True:
+            try:
+                self._poll_notifier = asyncio.Event()
+                await self._poll_notifier.wait()
+                await asyncio.sleep(1)
+                await self._update_balances()
+                await asyncio.sleep(1)
+                await self._update_order_status()
+
+            except asyncio.CancelledError:
+                raise
+            except Exception:
+                self.logger().network(
+                    "Unexpected error while fetching account updates.",
+                    exc_info=True,
+                    app_warning_msg=f"Could not fetch account updates on Bitfinex. "
+                )
+
+    async def _trading_rules_polling_loop(self):
+        """
+        Separate background process that periodically pulls for trading rule changes
+        (Since trading rules don't get updated often, it is pulled less often.)
+        """
+        while True:
+            try:
+                await safe_gather(self._update_trading_rules())
+                await asyncio.sleep(60)
+            except asyncio.CancelledError:
+                raise
+            except Exception:
+                self.logger().network(
+                    "Unexpected error while fetching trading rules.",
+                    exc_info=True,
+                    app_warning_msg=f"Could not fetch trading rule updates on Bitfinex. "
+                                    f"Check network connection."
+                )
+                await asyncio.sleep(0.5)
+
+    async def _api_balance(self):
+        path_url = "auth/r/wallets"
+        account_balances = await self._api_private("post", path_url=path_url, data={})
+        wallets = []
+        for balance_entry in account_balances:
+            wallets.append(Wallet._make(balance_entry[:5]))
+        return wallets
+
+    async def _api_platform_status(self):
+        path_url = "platform/status"
+        platform_status = await self._api_public("get", path_url=path_url)
+        return platform_status
+
+    async def _api_platform_config_pair_info(self):
+        path_url = "conf/pub:info:pair"
+        info = await self._api_public("get", path_url=path_url)
+        return info[0] if len(info) > 0 else 0
```

Comment:
I would default to return an empty `List`. Since subsequently `_format_trading_rules` expects a `List`